R/importSNV.TCGA.R

```diff
@@ -0,0 +1,323 @@
+            headValid <- FALSE
+        }
+        return(headValid)
+    }
+
+    # ==== Validata rSNV  =======================================
+
+    rSNVheadercheck <- function(file) {
+        headValid <- TRUE
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,1] == "sym") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,2] == "chr") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,3] == "start") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,4] == "end") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,5] == "strand") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,6] == "class") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,7] == "type") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,8] == "aRef") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,9] == "a1") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,10] == "a2") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,11] == "TCGA") {
+            headValid <- FALSE
+        }
+        if (! read.table(file, header = FALSE, fill = NA, stringsAsFactors = FALSE,
+                         na.strings = " ", nrows=1)[,12] == "UUID") {
+            headValid <- FALSE
+        }
+        return(headValid)
+    }
+
+    if (missing(rSNV)) {
+        rSNV <- "MAFtoSNV"
+
+        # add metdata
+        for (name in names(meta)) {
+            attr(rSNV, name) <- meta[[name]]
+        }
+
+        write("sym\tchr\tstart\tend\tstrand\tclass\ttype\taRef\ta1\ta2\tTCGA\tUUID",
+              rSNV, append=FALSE)
+
+        snvValidity <- TRUE
+    } else {
```

Comment:
You really don't need to read the entire old rSNV if it exists - you can just append to it after you confirm that the header is sane. use ```readLines()``` to read a single line, and check that.